dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -202,9 +209,12 @@ private LinkageChecker(
         classToSymbols.getClassToFieldSymbols();
     for (ClassFile classFile : classToFieldSymbols.keySet()) {
       ImmutableSet<FieldSymbol> fieldSymbols = classToFieldSymbols.get(classFile);
+      ImmutableSet<String> classFileNamesInSource =
+          classFile.getClassPathEntry().getClassFileNames();
       for (FieldSymbol fieldSymbol : fieldSymbols) {
-        if (!classFile.getClassPathEntry().getClassNames()
-            .contains(fieldSymbol.getClassBinaryName())) {
+        String classBinaryName = fieldSymbol.getClassBinaryName();
```

Comment:
This is the root cause of the problem. LinkageChecker was checking the existence of a class by class *binary* name (e.g., `pzrep.ReportServlet`), while the `ClassPathEntry.getClassNames()` returns a set of class *file* names (e.g., `WEB-INF.classes.pzrep.ReportServlet`).